File: Js_Solutions/capitalize/capitalize.js
// Write a function that accepts a string. The function
// should capitalize first letter of each word of the
// sentence and then return the capitalized string.

function capitalize(str) {
    let sentence = '';
    for (i = 0; i < str.length; i++) {
        if (i === 0 || str[i - 1] === ' ') {
            sentence += str[i].toUpperCase();
        } else {
            sentence += str[i];
        }
    }
    return sentence;
}

// Alternate solution
// function capitalize(str) {
//     const words = [];
//     str.split(' ').forEach(element => {
//         words.push(element[0].toUpperCase() + element.slice(1));
//     });

//     return words.join(' ');
// }

console.log(capitalize('hi! i am not coming!!'))

File: Python_Solutions/capitalize/capitalize.py
# Write a function that accepts a string. The function should
# capitalize first letter of each word of the sentence and
# then return the capitalized string.

def capitalize(string):
    sentence = ''
    for i in range(len(string)):
        if i == 0 or string[i-1] == ' ':
            sentence += string[i].upper()
        else:
            sentence += string[i]
    return sentence

# Alternate solution
# def capitalize(string):
#     words = []
#     for word in string.split(' '):
#         words.append(word[0].upper() + word[1:])
#     return (' ').join(words)


print(capitalize('look, it is working!'))
